src/helpers/shared/filters.js

```diff
@@ -0,0 +1,49 @@
+export const getFiltersFromUrl = (keys, defaultValues = {}) => {
+  const searchParams = new URLSearchParams(location.search);
+
+  let filters = {};
+
+  keys.forEach((key) => {
+    const values = searchParams.getAll(key);
+    filters = {
+      ...filters,
+      [key]: values.length > 1 ? values : values[0],
+    };
+  });
+
+  Object.keys(defaultValues).forEach((key) => {
+    const value = defaultValues[key];
+    filters = {
+      ...filters,
+      [key]: Array.isArray(filters[key])
+        ? [...new Set([...filters[key], ...(Array.isArray(value) ? value : [value])])]
+        : (value?.length > 0 && value) || filters[key],
+    };
+
+    if (Array.isArray(value)) {
+      value.forEach((item) => searchParams.getAll(key).includes(item) || searchParams.append(key, item));
+    } else {
+      searchParams.get(key) || (value && searchParams.set(key, value));
+    }
+  });
+
+  window.history.replaceState({}, '', `${location.pathname}?${searchParams.toString()}`);
```

Comment:
It would be better if we use the `react-router-dom` history instance. We can have a globally accessible history instance. Checkout these docs: https://reactrouter.com/web/api/Router/history-object